src/Rules/PhpDoc/IncompatiblePhpDocTypeRule.php

```diff
@@ -63,78 +65,107 @@ public function processNode(Node $node, Scope $scope): array
 		);
 		$nativeParameterTypes = $this->getNativeParameterTypes($node, $scope);
 		$nativeReturnType = $this->getNativeReturnType($node, $scope);
+		$byRefParameters = $this->getByRefParameters($node);
 
 		$errors = [];
 
-		foreach ($resolvedPhpDoc->getParamTags() as $parameterName => $phpDocParamTag) {
-			$phpDocParamType = $phpDocParamTag->getType();
-			if (!isset($nativeParameterTypes[$parameterName])) {
-				$errors[] = RuleErrorBuilder::message(sprintf(
-					'PHPDoc tag @param references unknown parameter: $%s',
-					$parameterName,
-				))->identifier('phpDoc.unknownParameter')->metadata(['parameterName' => $parameterName])->build();
+		foreach ([$resolvedPhpDoc->getParamTags(), $resolvedPhpDoc->getParamOutTags()] as $parameters) {
```

Comment:
the more I work on the PR, the more I get the feeling that the param-put phpdoc validation should be put into a separate rule.. turns out most of the checks here are not necessary,  because we don't need to validate subtype/supertype correctness ..?